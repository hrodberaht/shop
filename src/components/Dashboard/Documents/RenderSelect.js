import React from 'react';
import ShowErrorInForm from './ShowErrorInForm';

const RenderSelect = ({ input, children, meta, label }) => (
  <div>
    <label htmlFor={label}>{label}</label>
    <select className="invoice-select" {...input}>
      {children}
    </select>
    <ShowErrorInForm className="error-text" {...meta} />
  </div>
);

export default RenderSelect;
