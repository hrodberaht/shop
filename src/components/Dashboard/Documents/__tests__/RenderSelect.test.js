import React from 'react';
import RenderSelect from '../RenderSelect';

const defaultProps = {};

const setup = buildSetup(RenderSelect, defaultProps);

describe('<RenderSelect />', () => {
  describe('when initialized', () => {
    it('should render without crash', () => {
      const { wrapper } = setup();
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('when children props set', () => {
    const options = <option>5%</option>;

    it('should render children', () => {
      const { wrapper } = setup();
      wrapper.setProps({ children: options });
      expect(wrapper).toMatchSnapshot();
    });
  });
});
