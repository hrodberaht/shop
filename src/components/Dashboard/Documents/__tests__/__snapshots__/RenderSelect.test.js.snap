// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RenderSelect /> when children props set should render children 1`] = `
<div>
  <label />
  <select
    className="invoice-select"
  >
    <option>
      5%
    </option>
  </select>
  <ShowErrorInForm
    className="error-text"
    error={null}
    touched={false}
  />
</div>
`;

exports[`<RenderSelect /> when initialized should render without crash 1`] = `
<div>
  <label />
  <select
    className="invoice-select"
  />
  <ShowErrorInForm
    className="error-text"
    error={null}
    touched={false}
  />
</div>
`;
